inspirehep/modules/recordvalidators/validators/rules.py

```diff
@@ -0,0 +1,402 @@
+        required_values = ['proceedings', 'conference_paper']
+        if check_field_values_not_in_required_values_for_record(record, 'document_type', required_values):
+            raise ValidationError(key='globalErrors',
+                                  message=FIELD_REQUIRE_FIELD_VALUES_TEMPLATE.format(
+                                      field='cnum',
+                                      required='document_type',
+                                      values=required_values),
+                                  type='Warning')
+    return {}
+
+
+def check_accelerator_experiments_if_collaborations_exist(record):
+    if 'collaborations' in record:
+        if not 'accelerator_experiments' in record:
+            raise ValidationError(key='globalErrors',
+                                  message=FIELD_REQUIRE_FIELD_TEMPLATE.format(
+                                      field='collaborations',
+                                      required='accelerator_experiments'),
+                                  type='Warning')
+    return {}
+
+
+def check_if_reportnumber_exist_in_other_records(record):
+    try:
+        check_if_listfield_property_values_are_valid(record=record,
+                                                     field='report_numbers',
+                                                     prop='value',
+                                                     checker=query_db_for_duplicates_for_field_listitem_with_index,
+                                                     errortype='Warning')
+    except ValidationError:
+        raise
+    return {}
+
+
+def check_external_urls_if_work(record):
+    def _check_if_url_works(field, prop, value, index, errortype):
+        try:
+            resp = requests.get(value)
+            if not resp.ok:
+                raise ConnectionError
+        except ConnectionError:
+            raise ValidationError(key='/{field}/{index}/{prop}'.format(field=field, prop=prop, index=index),
+                                  message=FIELD_VALIDATION_TEMPLATE.format(
+                                      field='urls',
+                                      value=value
+                                  ),
+                                  type=errortype)
+        return {}
+
+    try:
+        check_if_listfield_property_values_are_valid(record=record,
+                                                     field='urls',
+                                                     prop='value',
+                                                     checker=_check_if_url_works,
+                                                     errortype='Warning'
+                                                     )
+    except ValidationError:
+        raise
+    return {}
+
+
+def check_external_dois_if_exist(record):
+    def _check_if_doi_works(field, prop, value, index, errortype):
+        doi_url = DOI_VALIDATION_URL.format(doi=value)
+        try:
+            resp = requests.get(doi_url)
```

Comment:
Double check this works with real values since there will be a redirect and I am not sure if `requests` library follows redirects by default.